app/forms/embargo_form.rb

```diff
@@ -8,54 +8,41 @@ class EmbargoForm
   attr_writer :embargoed_until
   attr_accessor :remove
 
-  validate :embargoed_until_is_valid_date,
-           unless: :remove?
-
   def initialize(work:, params:)
     @work = work
     super(params)
   end
 
   def embargoed_until
-    work_embargoed_until = if work.embargoed_until.respond_to?(:strftime)
-                             work.embargoed_until.strftime('%Y-%m-%d')
-                           end
-    @embargoed_until || work_embargoed_until
+    return work.embargoed_until if @embargoed_until.blank?
+
+    parsed_embargo_date
   end
 
   def remove?
     !!ActiveModel::Type::Boolean.new.cast(remove)
   end
 
   def save
-    return false unless valid?
-
     work.embargoed_until = if remove?
                              nil
                            elsif embargoed_until.present?
-                             Time.zone.parse(embargoed_until).beginning_of_day
+                             parsed_embargo_date
                            end
+
+    unless work.valid?
+      work.errors.each do |attr, message|
+        errors.add(attr, message)
+      end
```

Comment:
I haven't had a chance to fully review this yet, but one quick note here:

This code works, but it turns out that this way of iterating over errors is deprecated. The modern way to do this is
```ruby
work.errors.each do |error|
  errors.add(error.attribute, error.message)
end
```